ccp/lib/utils/command.py

```diff
@@ -0,0 +1,42 @@
+# -*- coding: utf-8 -*-
+from ccp.lib.exceptions import CommandOutputError
+import subprocess
+
+
+def run_command(cmd, shell=False, use_pipes=True):
+    """
+    Runs a shell command
+    :param cmd: The command that needs to be run.
+    :type cmd: s
+    :param shell: Default False: Whether to run raw shell commands with '|' and
+    redirections
+    :param use_pipes: Default False: If true, then pipes are used for
+    communication
+    :return: Command output and err
+    :raises subprocess.CalledProcessError
+    """
+    p = subprocess.Popen(
+        cmd, shell=shell, stdout=subprocess.PIPE, stderr=subprocess.PIPE
+    ) if use_pipes else subprocess.Popen(
```

Comment:
just make sure, the command which is going to return a lot of data (for eg, logs), we don't use the pipes as, it may chock up the available memory and block the command.

Reply:
I'd default on not using the pipe and when needed specify that explicitly.

Reply:
In most cases, i would agree with you. But here it is better this way.

In most usages in our code, we want to pipe the output. It is rare case where we just want to execute a command and dont care what it gives as output. That is why default is True.

It is for program efficiency.